server/src/main/java/org/apache/druid/catalog/model/TableMetadata.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import org.apache.druid.guice.annotations.PublicApi;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.jackson.JacksonUtils;
+
+import java.util.Objects;
+
+/**
+ * REST API level description of a table. Tables have multiple types
+ * as described by subclasses. Stores the operational aspects of a
+ * table, such as its name, creation time, state and spec.
+ *
+ * @see {@link ResolvedTable} for the semantic representation.
+ */
+@PublicApi
+public class TableMetadata
+{
+  public enum TableState
+  {
+    ACTIVE("A"),
```

Comment:
At present there are two states. I can't imagine adding many more, most likely zero. But, since there is some chance we will want another state, and since this is a DB field, we use the old "encode an enum as a single character" trick. The column itself is (or should be) a `CHAR(1)`. As a result, there an be no states with more than one letter.